src/Compilers/CSharp/Portable/Symbols/Source/SourceConstructorSymbol.cs

```diff
@@ -36,6 +36,12 @@ internal sealed class SourceConstructorSymbol : SourceMethodSymbol
             var declarationModifiers = this.MakeModifiers(syntax.Modifiers, methodKind, location, diagnostics, out modifierErrors);
             this.MakeFlags(methodKind, declarationModifiers, returnsVoid: true, isExtensionMethod: false);
 
+            if (syntax.Identifier.ValueText != containingType.Name)
```

Comment:
we probably need a test that validates that unicode-escapes are working properly.